apps/console/src/features/organizations/components/edit-organization-role/organization-role-permission.tsx

```diff
@@ -0,0 +1,291 @@
+            onSubmit,
+            roleObject,
+            isEdit,
+            initialValues,
+            isRole,
+            isSubmitting,
+            permissionsToHide,
+            ["data-testid"]: testId
+        } = props;
+
+        const { t } = useTranslation();
+
+        const [ permissions, setPermissions ] = useState<TreeNode[]>([]);
+        const [ checkedPermission, setCheckedPermission ] = useState<TreeNode[]>([]);
+        const [ previouslyCheckedKeys, setPreviouslyCheckedKeys ] = useState<string[]>([]);
+        const [ defaultExpandedKeys, setDefaultExpandKeys ] = useState<string[]>([]);
+        const [ isPermissionsLoading, setIsPermissionsLoading ] = useState<boolean>(true);
+
+        useEffect(() => {
+            const checkedNodes: TreeNode[] = [];
+
+            OrganizationRoleManagementUtils.getAllOrganizationLevelPermissions(permissionsToHide)
+                .then((permissionTree: TreeNode[]) => {
+                    setPermissions(permissionTree);
+                    setDefaultExpandKeys([ permissionTree[0].key.toString() ]);
+                    setIsPermissionsLoading(false);
+                });
+
+            if (initialValues && initialValues.length > 0) {
+                const previousFormCheckedKeys: string[] = [];
+
+                initialValues.forEach(initialKey => {
+                    previousFormCheckedKeys.push(initialKey.key.toString());
+                });
+                setPreviouslyCheckedKeys(previousFormCheckedKeys);
+                previouslyCheckedKeys?.forEach(key => {
+                    checkedNodes.push(getNodeByKey(key, permissions));
+                });
+                setCheckedPermission(checkedNodes);
+            }
+
+            if (isRole && roleObject) {
+                setPreviouslyCheckedKeys(roleObject.permissions);
+                previouslyCheckedKeys?.forEach(key => {
+                    checkedNodes.push(getNodeByKey(key, permissions));
+                });
+                setCheckedPermission(checkedNodes);
+            }
+        }, [ permissions.length > 0 ]);
+
+        /**
+         * Util function to disable checking of a node and it's children.
+         *
+         * @param permissionNodes - array of permission nodes
+         * @param state - disable state
+         */
+        const disableTreeNode = (permissionNodes: TreeNode[], state: boolean) => {
+            permissionNodes.forEach((permission: TreeNode) => {
+                permission.disableCheckbox = state;
+                if (permission.children) {
+                    disableTreeNode(permission.children, state);
+                }
+            });
+        };
+
+        /**
+         * Event handler when a node is checked on the permission tree.
+         *
+         * @param checked - checked states of the node
+         * @param info - checked information
+         */
+        const onCheck = (checked: { checked: React.ReactText[]; halfChecked: React.ReactText[] }, info) => {
```

Comment:
shall we type `info` param and the return type

Reply:
Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/630fa8dda5d13075f2d424b2493249572e255217